File: About_me/MainForm.rb
require "mscorlib"
require "System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089"
require "System.Drawing, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"

class MainForm < System::Windows::Forms::Form
	def initialize()
		self.InitializeComponent()
	end

	def InitializeComponent()
		@button1 = System::Windows::Forms::Button.new()
		@button2 = System::Windows::Forms::Button.new()
		@label1 = System::Windows::Forms::Label.new()
		self.SuspendLayout()
		# 
		# button1
		# 
		@button1.BackColor = System::Drawing::Color.NavajoWhite
		@button1.Location = System::Drawing::Point.new(211, 175)
		@button1.Name = "button1"
		@button1.Size = System::Drawing::Size.new(172, 83)
		@button1.TabIndex = 0
		@button1.Text = "Show"
		@button1.UseVisualStyleBackColor = false
		@button1.Click { |sender, e| self.Button1Click(sender, e) }
		# 
		# button2
		# 
		@button2.BackColor = System::Drawing::Color.NavajoWhite
		@button2.Location = System::Drawing::Point.new(398, 175)
		@button2.Name = "button2"
		@button2.Size = System::Drawing::Size.new(172, 83)
		@button2.TabIndex = 1
		@button2.Text = "Clear"
		@button2.UseVisualStyleBackColor = false
		@button2.Click { |sender, e| self.Button2Click(sender, e) }
		# 
		# label1
		# 
		@label1.BackColor = System::Drawing::Color.LemonChiffon
		@label1.Location = System::Drawing::Point.new(211, 74)
		@label1.Name = "label1"
		@label1.Size = System::Drawing::Size.new(359, 98)
		@label1.TabIndex = 2
		# 
		# MainForm
		# 
		self.BackColor = System::Drawing::Color.Bisque
		self.ClientSize = System::Drawing::Size.new(826, 303)
		self.Controls.Add(@label1)
		self.Controls.Add(@button2)
		self.Controls.Add(@button1)
		self.Name = "MainForm"
		self.Text = "About_me"
		self.ResumeLayout(false)
	end

	def Button2Click(sender, e)
		@label1.Text = ""
	end

	def Button1Click(sender, e)
		@label.Text = "My name is Owen White"
	end
end

File: Prog122b/MainForm.py
import System.Drawing
import System.Windows.Forms

from System.Drawing import *
from System.Windows.Forms import *

class MainForm(Form):
	def __init__(self):
		self.InitializeComponent()
	
	def InitializeComponent(self):
		self._listBox1 = System.Windows.Forms.ListBox()
		self._button1 = System.Windows.Forms.Button()
		self._button2 = System.Windows.Forms.Button()
		self._button3 = System.Windows.Forms.Button()
		self._listBox2 = System.Windows.Forms.ListBox()
		self._label1 = System.Windows.Forms.Label()
		self._label2 = System.Windows.Forms.Label()
		self.SuspendLayout()
		# 
		# listBox1
		# 
		self._listBox1.BackColor = System.Drawing.SystemColors.ActiveBorder
		self._listBox1.Font = System.Drawing.Font("Perpetua", 20.25, System.Drawing.FontStyle.Bold | System.Drawing.FontStyle.Italic, System.Drawing.GraphicsUnit.Point, 0)
		self._listBox1.FormattingEnabled = True
		self._listBox1.ItemHeight = 31
		self._listBox1.Location = System.Drawing.Point(12, 43)
		self._listBox1.Name = "listBox1"
		self._listBox1.Size = System.Drawing.Size(381, 159)
		self._listBox1.TabIndex = 0
		# 
		# button1
		# 
		self._button1.BackColor = System.Drawing.SystemColors.ActiveBorder
		self._button1.Font = System.Drawing.Font("Perpetua", 20.25, System.Drawing.FontStyle.Bold | System.Drawing.FontStyle.Italic, System.Drawing.GraphicsUnit.Point, 0)
		self._button1.Location = System.Drawing.Point(12, 217)
		self._button1.Name = "button1"
		self._button1.Size = System.Drawing.Size(322, 80)
		self._button1.TabIndex = 1
		self._button1.Text = "Show List"
		self._button1.UseVisualStyleBackColor = False
		self._button1.Click += self.Button1Click
		# 
		# button2
		# 
		self._button2.BackColor = System.Drawing.SystemColors.ActiveBorder
		self._button2.Font = System.Drawing.Font("Perpetua", 20.25, System.Drawing.FontStyle.Bold | System.Drawing.FontStyle.Italic, System.Drawing.GraphicsUnit.Point, 0)
		self._button2.Location = System.Drawing.Point(340, 217)
		self._button2.Name = "button2"
		self._button2.Size = System.Drawing.Size(110, 80)
		self._button2.TabIndex = 2
		self._button2.Text = "Clear"
		self._button2.UseVisualStyleBackColor = False
		self._button2.Click += self.Button2Click
		# 
		# button3
		# 
		self._button3.BackColor = System.Drawing.SystemColors.ActiveBorder
		self._button3.Font = System.Drawing.Font("Perpetua", 20.25, System.Drawing.FontStyle.Bold | System.Drawing.FontStyle.Italic, System.Drawing.GraphicsUnit.Point, 0)
		self._button3.Location = System.Drawing.Point(456, 217)
		self._button3.Name = "button3"
		self._button3.Size = System.Drawing.Size(334, 80)
		self._button3.TabIndex = 3
		self._button3.Text = "Exit"
		self._button3.UseVisualStyleBackColor = False
		self._button3.Click += self.Button3Click
		# 
		# listBox2
		# 
		self._listBox2.BackColor = System.Drawing.SystemColors.ActiveBorder
		self._listBox2.Font = System.Drawing.Font("Perpetua", 20.25, System.Drawing.FontStyle.Bold | System.Drawing.FontStyle.Italic, System.Drawing.GraphicsUnit.Point, 0)
		self._listBox2.FormattingEnabled = True
		self._listBox2.ItemHeight = 31
		self._listBox2.Location = System.Drawing.Point(399, 43)
		self._listBox2.Name = "listBox2"
		self._listBox2.Size = System.Drawing.Size(391, 159)
		self._listBox2.TabIndex = 4
		# 
		# label1
		# 
		self._label1.BackColor = System.Drawing.Color.DimGray
		self._label1.Font = System.Drawing.Font("Perpetua Titling MT", 9.75, System.Drawing.FontStyle.Bold | System.Drawing.FontStyle.Italic, System.Drawing.GraphicsUnit.Point, 0)
		self._label1.ForeColor = System.Drawing.SystemColors.ButtonHighlight
		self._label1.Location = System.Drawing.Point(12, 9)
		self._label1.Name = "label1"
		self._label1.Size = System.Drawing.Size(381, 31)
		self._label1.TabIndex = 5
		self._label1.Text = "Hours:"
		# 
		# label2
		# 
		self._label2.BackColor = System.Drawing.Color.DimGray
		self._label2.Font = System.Drawing.Font("Perpetua Titling MT", 9.75, System.Drawing.FontStyle.Bold | System.Drawing.FontStyle.Italic, System.Drawing.GraphicsUnit.Point, 0)
		self._label2.ForeColor = System.Drawing.SystemColors.ButtonHighlight
		self._label2.Location = System.Drawing.Point(399, 9)
		self._label2.Name = "label2"
		self._label2.Size = System.Drawing.Size(391, 31)
		self._label2.TabIndex = 6
		self._label2.Text = "Pay in Dollars:"
		# 
		# MainForm
		# 
		self.BackColor = System.Drawing.Color.Maroon
		self.ClientSize = System.Drawing.Size(802, 309)
		self.Controls.Add(self._label2)
		self.Controls.Add(self._label1)
		self.Controls.Add(self._listBox2)
		self.Controls.Add(self._button3)
		self.Controls.Add(self._button2)
		self.Controls.Add(self._button1)
		self.Controls.Add(self._listBox1)
		self.Name = "MainForm"
		self.Text = "Prog122b"
		self.ResumeLayout(False)


	def Button1Click(self, sender, e):
		for num in range (1, 40+1):
			self._listBox1.Items.Add(num)
		for num in range (1, 40+1):
			self._listBox2.Items.Add(num * 4)
			

	def Button2Click(self, sender, e):
		self._listBox1.Items.Clear()
		self._listBox2.Items.Clear()
		
	def Button3Click(self, sender, e):
		Application.Exit()